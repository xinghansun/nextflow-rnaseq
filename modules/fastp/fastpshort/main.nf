#! /usr/bin/env nextflow

process fastp_short {
	tag "${sample_id}"
	publishDir "result/fastp/${sample_id}"

	conda "${moduleDir}/environment.yml"

	input: 
	tuple val(sample_id), path(reads)

	output:
	tuple val(sample_id), path("*.fastp.fastq.gz"), emit: trimmed_reads
	tuple val(sample_id), path("${sample_id}.html"), emit: html
	tuple val(sample_id), path("${sample_id}.json"), emit: json

	script:
	def o = reads.size() == 2 ? "${sample_id}_R1.fastp.fastq.gz" : "${sample_id}.fastp.fastq.gz"
	def paired_syntax = reads.size() == 2 ? "-I ${reads[1]} -O ${sample_id}_R2.fastp.fastq.gz" : ""

	"""
	echo "${moduleDir}/environment.yml"
	fastp \\
		-i ${reads[0]} \\
		-o ${o} \\
		-h ${sample_id}.html \\
		-j ${sample_id}.json \\
		${paired_syntax} \\
		--thread ${task.cpus}
	"""

}