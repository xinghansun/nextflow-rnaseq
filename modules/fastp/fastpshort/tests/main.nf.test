nextflow_process {

    name "Test Process fastp_short"
    script "../main.nf"
    process "fastp_short"

    test("Test single end inputs") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                input[0] = Channel.of([
                    "ENCSR000COQ1", 
                    file("${projectDir}/data/tests/ENCSR000COQ1_1.fastq.gz")
                    ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.trimmed_reads).match()
        }

    }

    test("Test paired end inputs") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                input[0] = Channel.of([
                    "ENCSR000COQ1", 
                    [
                        file("${projectDir}/data/tests/ENCSR000COQ1_1.fastq.gz"),
                        file("${projectDir}/data/tests/ENCSR000COQ1_2.fastq.gz"),
                    ]
                    ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.trimmed_reads).match()
        }

    }

}
