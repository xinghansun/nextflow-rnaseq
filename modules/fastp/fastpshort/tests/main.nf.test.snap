{
    "Test paired end inputs": {
        "content": [
            [
                [
                    "ENCSR000COQ1",
                    [
                        "ENCSR000COQ1_R1.fastp.fastq.gz:md5,dc8939a37ac93f426d1309502f95c4d6",
                        "ENCSR000COQ1_R2.fastp.fastq.gz:md5,cf3baec2839ea0f36420e314d6ed143e"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-25T02:18:09.818026"
    },
    "Test single end inputs": {
        "content": [
            [
                [
                    "ENCSR000COQ1",
                    "ENCSR000COQ1.fastp.fastq.gz:md5,49d498ed2f2b4f561d84bc0a3b72cf02"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-25T02:17:05.646137"
    }
}